src/main/java/babysitter/BabySitter.java

```diff
@@ -33,50 +30,34 @@ boolean invalidUserEntry(int startTime, int endTime, int bedTime) {
 				if (invalids[i] == sc.nextInt()) {
 					invalidTracker = true;
 				}
-				invalidTracker = false;
 			}
 		}
 		return invalidTracker;
 
 	}
 
 	int convertAm(int am) {
-		if (am > 0 && am <= 4) {
-			am += 24;
-		}
-		return am;
+		return (am <= 4) ? am + 24 : am;
 	}
 
 	int calculatePay() {
-		return hoursBeforeBedtime() * BEFORE_BEDTIME_RATE
-				+ hoursBetweenBedtimeAndMidnight() * BETWEEN_BEDTIME_AND_MIDNIGHT_RATE
-				+ hoursAfterMidnight() * AFTER_MIDNIGHT_RATE;
-	}
-
-	int hoursBeforeBedtime() {
-		if (startTime >= 17 && endTime <= bedTime) {
-			return endTime - startTime;
-		} else {
+		int result = 0;
+		
+		if (!isValidWorkHours(startTime, endTime)) {
 			return 0;
 		}
-	}
-
-	int hoursBetweenBedtimeAndMidnight() {
-		if (startTime <= bedTime && endTime > bedTime && endTime <= 24) {
-			return endTime - bedTime;
-		} else
-			return 0;
-	}
-
-	int hoursAfterMidnight() {
-		if (startTime > 24 && endTime > 24) {
-			return endTime - startTime;
-		} else if (endTime > 24) {
-			return endTime - 24;
-		} else {
-			return 0;
+		
+		for (int i = startTime; i < endTime; i++) {
+			if (i < bedTime && i < 24) {
+				result += BEFORE_BEDTIME_RATE;
+			} else if (i >= bedTime && i < 24) {
+				result += BETWEEN_BEDTIME_AND_MIDNIGHT_RATE;
+			} else {
+				result += AFTER_MIDNIGHT_RATE;
+			}
 		}
-
+		
+		return result;
```

Comment:
This diff looks a little goofy on Github, but basically what I did was boil the math down to this:

```
int calculatePay() {
	int result = 0;
		
	if (!isValidWorkHours(startTime, endTime)) {
		return 0;
	}
		
	for (int i = startTime; i < endTime; i++) {
		if (i < bedTime && i < 24) {
			result += BEFORE_BEDTIME_RATE;
		} else if (i >= bedTime && i < 24) {
			result += BETWEEN_BEDTIME_AND_MIDNIGHT_RATE;
		} else {
			result += AFTER_MIDNIGHT_RATE;
		}
	}
		
	return result;
}
```

With a solution like this, the `hours...()` methods are no longer necessary, and now the tests pass.

For the most part, those tests were failing because there were some functional issues with your `hours...()` methods. For example, in `hoursBeforeBedtime()`, your if condition is `startTime >= 17 && endTime <= bedTime`, but `endTime` will usually not be before `bedTime`, so in most cases that method was returning `0` instead of doing the math.